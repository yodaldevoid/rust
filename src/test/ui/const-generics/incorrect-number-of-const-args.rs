#![feature(const_generics)]
//~^ WARN the feature `const_generics` is incomplete and may cause the compiler to crash

fn foo<const X: usize>() -> usize {
    0
}

fn main() {
    foo(); //~ ERROR too few const parameters provided
    foo::<0, 0>(); //~ ERROR too many const parameters provided
}
