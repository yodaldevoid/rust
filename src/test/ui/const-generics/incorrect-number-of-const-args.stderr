warning: the feature `const_generics` is incomplete and may cause the compiler to crash
  --> $DIR/const-param-from-outer-fn.rs:1:12
   |
LL | #![feature(const_generics)]
   |            ^^^^^^^^^^^^^^

error[E0086]: too few const parameters provided: expected 1 const parameter, found 0 const parameters
  --> $DIR/incorrect-number-of-const-args.rs:14:5
   |
LL |     foo(); //~ ERROR too few const parameters provided
   |     ^^^ expected 1 const parameter

error[E0085]: too many const parameters provided: expected at most 1 const parameter, found 2 const parameters
  --> $DIR/incorrect-number-of-const-args.rs:15:14
   |
LL |     foo::<0, 0>(); //~ ERROR too many const parameters provided
   |              ^ expected 1 const parameter

error: aborting due to 2 previous errors

Some errors occurred: E0085, E0086.
For more information about an error, try `rustc --explain E0085`.
